### PR TITLE
Support resizing and cropping native images

### LINKED ISSUE #6523: Native image resize and crop
Is it possible to expose Chromiums image manipulation functionality to Electron?

I've found image resizing on node is troublesome to do cross-platform and requires compiling of native modules. Meanwhile Chromium already supports a whole host of image formats.

I created [an electron module](https://github.com/davej/electron-image-resize) to do this. It essentially opens the source image in a new browser window with styles set to `img { width: 100%; height: 100%; }` and then captures the page (`win.capturePage`). It works, but the performance leaves a lot to be desired which makes it impractical for a lot of scenarios, it would be much better if there was some way to get at chromium's image manipulation.

* Electron version: 1.2.6
* Operating system: All

### PR DESCRIPTION
**Pull request by kevinsawicki**
This pull requests add cropping and resizing support to native images.

- [x] Add `resize` method
- [x] Add `crop` method
- [x] Add `getAspectRatio` method 
- [x] Add tests
- [x] Add docs

Closes #6523

### CODE REVIEW

**Comment by davej** [comment]
Awesome. 👍

(Something small I noticed… the type signature for `nativeImage.getSize()` is incorrect, it should be an Object. Do I just PR the english docs to fix it or are the type signatures now being generated from somewhere?)

**Comment by kevinsawicki** [comment]
> Do I just PR the english docs to fix it or are the type signatures now being generated from somewhere?

PR to `docs/api/native-image.md` should be good 👍 Thanks.

File: docs/api/native-image.md [hunk 1]
```diff
@@ -215,4 +215,28 @@ Marks the image as a template image.
 
 Returns `Boolean` - Whether the image is a template image.
 
+#### `image.crop(region)`
+
+* `region` Object - The region of the image to crop
```
Comment (zcbenz) [inline docs/api/native-image.md]: `rect` would be a better name, `region` is usually an union of `rect`s.

**Comment by zcbenz** [comment]
👍

**Comment by davej** [comment]
Just a thought… if I do an image resize with one param:

```js
image.resize({ width: 200 });
```

Then I would have probably expected the `height` to scale proportionately with the `width` (rather than stay the same). It's rare that you would actually want to skew the image, the more common case is that you want to scale the image to a fixed width or height. Of course this can be done in userland using javascript and `image.getSize()` so it's not a big issue. Might be worth mentioning it in the docs though.

**Comment by kevinsawicki** [comment]
> Then I would have probably expected the height to scale proportionately with the width (rather than stay the same)

Good call, I've updated this pull request to now have that behavior. Setting only `height` or `width` will use a preserved aspect ratio value for the other.

Also added a `getAspectRatio` API since that might be useful as well when resizing/cropping.

**Comment by zcbenz** [comment]
👍

**Pull request merged**